app/uk/gov/hmrc/plasticpackagingtax/registration/controllers/group/GroupMemberGrsController.scala

```diff
@@ -89,23 +119,39 @@ class GroupMemberGrsController @Inject() (
   ): Future[Either[ServiceError, Registration]] =
     for {
       details <- getDetails(journeyId)
-      result <- update { registration =>
-        val updatedGroupDetails: GroupDetail = registration.groupDetail match {
-          case Some(groupDetail) =>
-            val members: Seq[GroupMember] = groupDetail.members :+ addGroupMember(details, orgType)
-            groupDetail.copy(members = members, currentMemberOrganisationType = None)
-          case None => throw new InternalServerException(s"No group detail")
-        }
-        registration.copy(groupDetail = Some(updatedGroupDetails))
-      }
+      result  <- updateGroupDetails(orgType, details)
     } yield result
 
   private def addGroupMember(details: IncorporationDetails, orgType: OrgType): GroupMember =
     GroupMember(customerIdentification1 = details.companyNumber,
                 customerIdentification2 = Some(details.ctutr),
                 organisationDetails =
-                  Some(OrganisationDetails(orgType.toString, details.companyName)),
+                  Some(
+                    OrganisationDetails(orgType.toString,
+                                        details.companyName,
+                                        details.registration.registeredBusinessPartnerId
+                    )
+                  ),
                 addressDetails = details.companyAddress.toGroupAddressDetails
     )
 
+  private def updateGroupDetails(orgType: OrgType, details: IncorporationDetails)(implicit
+    hc: HeaderCarrier,
+    request: JourneyRequest[_]
+  ): Future[Either[ServiceError, Registration]] =
+    update { registration =>
+      val updatedGroupDetails: GroupDetail = registration.groupDetail match {
+        case Some(groupDetail) =>
+          val member = addGroupMember(details, orgType)
+          if (!groupDetail.memberAlreadyPresent(member)) {
+            val members: Seq[GroupMember] = groupDetail.members :+ member
+            groupDetail.copy(members = members, currentMemberOrganisationType = None)
```

Comment:
I wonder whether the resetting of `currentMemberOrganisationType` needs to also be done in unhappy path scenarios, like the member already existing?

Reply:
I think the unhappy path currently I am redirecting to OrganisationListController which will override `currentMemberOrganisationType` the value when the user selects org type

Reply:
You might have to talk me through this one. I'm concerned that there might be scenarios where `currentMemberOrganisationType` remains set and will therefore result in the entity type being preset if an attempt to add another group member is attempted.

Reply:
Yeah sure, we can talk through this.